sunpy/net/fido_factory.py

```diff
@@ -361,10 +391,19 @@ def fetch(self, *query_results, path=None, max_conn=5, progress=True,
 
         reslist = []
         for query_result in query_results:
-            for block in query_result.responses:
-                reslist.append(block.client.fetch(block, path=path,
-                                                  downloader=downloader,
-                                                  wait=False, **kwargs))
+            if isinstance(query_result, BaseQueryResponse):
+                result = query_result.client.fetch(query_result, path=path,
+                                                   downloader=downloader,
+                                                   wait=False, **kwargs)
+                if result is not NotImplemented:
+                    reslist.append(result)
+            else:
+                for block in query_result.responses:
+                    result = block.client.fetch(block, path=path,
+                                                downloader=downloader,
+                                                wait=False, **kwargs)
+                    if result is not NotImplemented:
+                        reslist.append(result)
```

Comment:
This block seems repetitive. Why don't you construct a list and then iterate over it even if it only has one element? This would save you duplicating the fetch and NotImplemented code?

Reply:
I have fixed this in a way you will not like.